Explain the details to quota compensation (#981)
harvester/docs: docs/rancher/resource-quota.md

You must set the annotation before the migration starts. If the annotation is set while the values are already being adjusted, Harvester is unable to automatically restore the previous configuration.

:::

> ## ResourceQuota Compensation During Migration

Reviewer: ```suggestion
## ResourceQuota Automatic Adjustment During Migration
```
here and elsewhere, consider using "adjustment" instead of "compensation"; the latter is more commonly used in the context of employment, insurance, lawsuit etc.

_Available as of v1.8.0_

> If you adjust the [VM Overhead Memory](#overhead-memory-of-virtual-machine) cluster settings while a Virtual Machine (VM) is already running, the VM's memory footprint may increase. This can cause issues during Live Migration, as the target node must instantiate a new VM instance that requires more memory than the original.

Reviewer: ```suggestion
If you adjust the [VM Overhead Memory](#overhead-memory-of-virtual-machine) cluster settings while a Virtual Machine (VM) is already running, the VM's memory footprint may increase. This can cause issues during live migration, as the target node must instantiate a new VM instance that requires more memory than the original.
```

> When the destination Namespace has a strict ResourceQuota, the migration may be blocked because the cumulative memory usage of both the source and target VMs temporarily exceeds the quota even when the quota has already been scaled up automatically.

Reviewer: ```suggestion
When the destination namespace has a strict ResourceQuota, the migration may be blocked because the cumulative memory usage of both the source and target VMs temporarily exceeds the quota even when the quota has already been scaled up automatically.
```

> ### Automatic Resolution

Reviewer: ```suggestion
```
probably don't need this title if you just add the word "Automatic" to the main heading as suggested above

Harvester automatically manages this bottleneck through the following workflow:

**Detection**: Harvester identifies when a migration is specifically blocked by ResourceQuota limitations.

> **Delta Compensation**: The system automatically injects a temporary "quota compensation" (the delta between the current limit and current usage plus the migration requirements) to allow the migration to proceed.

Reviewer: ```suggestion
**Delta Compensation**: The system automatically calculates a temporary "quota compensation" (the delta between the current limit and current usage plus the migration requirements) to allow the migration to proceed.
```

> **Cleanup**: Once the migration is complete (whether it succeeded or failed), Harvester removes the temporary compensation, returning the ResourceQuota to its original state.

Reviewer: ```suggestion
**Cleanup**: Once the migration is completed (whether it succeeded or failed), Harvester removes the temporary compensation, returning the ResourceQuota to its original state.
```

:::info important

> - This feature acts as an automatic workaround, ensuring that global policy changes (like memory overhead adjustments) do not accidentally "lock" your VMs to their current nodes.

Reviewer: ```suggestion
- This automated feature prevents global policy changes like memory overhead adjustments from accidentally blocking VMs live migration.
```

> - The compensation applies only to already running VMs. If a running VM is stopped and then restarted, its resource allocation is strictly governed by the original quota. As a result, a VM that was previously running may fail to "cold boot" if the overhead is increased while the ResourceQuota remains unchanged.

Reviewer: ```suggestion
- The compensation applies only to already running VMs. If a running VM is stopped and then restarted, no automatic quota adjustment will be triggered. As a result, the VM may fail to start if its overhead requirement exceeds the associated ResourceQuota.
```
